nixos/modules/services/hardware/fwupd.nix

```diff
@@ -16,6 +16,7 @@ let
     "fwupd/fwupd.conf" = {
       source = format.generate "fwupd.conf" {
         fwupd = cfg.daemonSettings;
+      } // lib.optionalAttrs (lib.length (lib.attrNames cfg.uefiCapsuleSettings) != 0) {
```

Comment:
This silently breaks uefiCapsuleSettings entirely because of operator precedence, since it gets eval'd the same as:

```nix
      source = (format.generate "fwupd.conf" {
        fwupd = cfg.daemonSettings;
      }) // lib.optionalAttrs (lib.length (lib.attrNames cfg.uefiCapsuleSettings) != 0) {
        uefi_capsule = cfg.uefiCapsuleSettings;
      };
```

instead of the intended:

```nix
      source = format.generate "fwupd.conf" ({
        fwupd = cfg.daemonSettings;
      } // lib.optionalAttrs (lib.length (lib.attrNames cfg.uefiCapsuleSettings) != 0) {
        uefi_capsule = cfg.uefiCapsuleSettings;
      });
```

I'm preparing a PR to fix it now

Reply:
Ouch! Thanks for fixing that!